src/data-factory.ts

```diff
@@ -0,0 +1,11 @@
+import dataFactory from '@rdfjs/data-model';
+import createDataset from 'rdf-dataset-indexed';
+import { ExtendedDataFactory } from './middleware/dataset';
+
+const extendedDataFactory: ExtendedDataFactory = { ...dataFactory, dataset: createDataset };
```

Comment:
a method name like `dataset()` could make it look like a singleton, whereas it's a factory that produces a new dataset each time it is called

Reply:
It's https://rdf.js.org/dataset-spec/#datasetcorefactory-interface.